compiler/qsc_frontend/src/resolve.rs

```diff
@@ -64,6 +64,19 @@ pub(super) enum Error {
         second_open_span: Span,
     },
 
+    #[error("`{name}` could refer to the item in `{candidate_a}` or an item in `{candidate_b}`")]
+    #[diagnostic(help(
+        "a namespace should not be declared with the same name as a namespace in the prelude"
+    ))]
+    #[diagnostic(code("Qsc.Resolve.AmbiguousPrelude"))]
+    AmbiguousPrelude {
+        name: String,
+        candidate_a: String,
+        candidate_b: String,
+        #[label]
+        span: Span,
```

Comment:
Not super necessary but I think we might as well keep `Ambiguous` and `AmbiguousPrelude` as similar as possible:

```suggestion
        #[label("ambiguous name")]
        span: Span,
```